packages/dds/task-manager/src/taskManager.ts

```diff
@@ -254,18 +256,18 @@ export class TaskManager extends SharedObject<ITaskManagerEvents> implements ITa
         this.latestPendingOps.set(taskId, pendingOp);
     }
 
-    public async lockTask(taskId: string) {
+    public async volunteerForTask(taskId: string) {
```

Comment:
I wonder if it would be nice to make volunteerForTask robust to disconnection blips while queued.  E.g. still only resolve when we reach the head of the queue or are permanently ejected from the queue, but if I get bumped due to a connection blip it automatically puts me back in line on reconnect?

Reply:
That sounds like a good idea. However, we should probably still reject if the disconnection is long enough. I'm not sure exactly what the cut off is, but maybe something like 3 seconds?

Reply:
My preference is to never bake in a timeout - we don't know whether a timeout is appropriate for a customer's scenario or if so, how long.  If a customer wants to implement their own timeout, they have the tools they need to do so.

Also feel free to defer this particular feedback to another PR to avoid scope creep, was just thinking about it.